Firebase/Database/Api/FIRDatabase.m

```diff
@@ -92,27 +99,59 @@ + (FIRDatabase *)database {
     return [FIRDatabase databaseForApp:app];
 }
 
++ (FIRDatabase *)databaseWithURL:(NSString *)url {
+    FIRApp *app = [FIRApp defaultApp];
+    if (app == nil) {
+        [NSException raise:@"FIRAppNotConfigured"
+                    format:@"Failed to get default Firebase Database instance. "
+                           @"Must call `[FIRApp configure]` (`FirebaseApp.configure()` in Swift) "
+                           @"before using Firebase Database."];
+    }
+    return [FIRDatabase databaseForApp:app URL:url];
+}
+
 + (FIRDatabase *)databaseForApp:(FIRApp *)app {
     if (app == nil) {
         [NSException raise:@"InvalidFIRApp" format:@"nil FIRApp instance passed to databaseForApp."];
     }
-    NSMutableDictionary *instances = [self instances];
-    @synchronized (instances) {
-        FIRDatabase *database = instances[app.name];
-        if (!database) {
-            NSString *databaseUrl = app.options.databaseURL;
-            if (databaseUrl == nil) {
-                [NSException raise:@"MissingDatabaseURL" format:@"Failed to get FIRDatabase instance: FIRApp object has no "
-                        "databaseURL in its FirebaseOptions object."];
-            }
 
-            FParsedUrl *parsedUrl = [FUtilities parseUrl:databaseUrl];
-            if (![parsedUrl.path isEmpty]) {
-                [NSException raise:@"InvalidDatabaseURL" format:@"Configured Database URL '%@' is invalid. It should "
-                                                                        "point to the root of a Firebase Database but it includes a path: %@",
-                                                                databaseUrl, [parsedUrl.path toString]];
-            }
+    return [FIRDatabase databaseForApp:app URL:app.options.databaseURL];
+}
 
++ (FIRDatabase *)databaseForApp:(FIRApp *)app URL:(NSString *)url {
+    if (app == nil) {
+        [NSException raise:@"InvalidFIRApp"
+                    format:@"nil FIRApp instance passed to databaseForApp."];
+    }
+
+    if (url == nil) {
+        [NSException raise:@"MissingDatabaseURL"
+                    format:@"Failed to get FirebaseDatabase instance: "
+                            "Specify DatabaseURL within FIRApp or from your databaseForApp: call."];
```

Comment:
should this be`databaseForApp:URL: call` ?  (and below)

Reply:
Yes